packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -210,30 +213,37 @@ class VisualStudio {
     _cachedUsableVisualStudioDetails ??= _visualStudioDetails(
         requiredComponents: _requiredComponents().keys,
         additionalArguments: <String>[_prereleaseKey]);
-    if (_cachedUsableVisualStudioDetails != null) {
-      if (installationHasIssues(_cachedUsableVisualStudioDetails)) {
+    // If an installation with required components if found, but it has issues (e.g. is not complete)
+    // or none is found, assign a sentinel empty value to avoid querying vswhere again.
```

Comment:
This is essentially duplicating the note a few lines above; remove one or the other copy.